### ISSUE TITLE
false positive too-many-ancestors if class inherits MutableMapping

### ISSUE THREAD

**Issue by kiefermat** [PyCQA/pylint#4166]
### Steps to reproduce

Given a file `a.py`:
```python
# pylint: disable=missing-docstring

from typing import Iterator, MutableMapping, Optional


class PylintTestMap(MutableMapping[str, str]):
    config: MutableMapping[str, str] = dict()

    def __getitem__(self, item: str) -> Optional[str]:
        return self.config.get(item)

    def __len__(self) -> int:
        return len(self.config)

    def __iter__(self) -> Iterator[str]:
        return iter(self.config)

    def __setitem__(self, key: str, value: str) -> None:
        self.config[key] = value

    def __delitem__(self, key: str) -> None:
        del self.config[key]
```

### Current behavior

Result of `pylint a.py`:
```
************* Module a
a.py:7:0: R0901: Too many ancestors (8/7) (too-many-ancestors)

------------------------------------------------------------------
Your code has been rated at 9.23/10 (previous run: 9.23/10, +0.00)
```

### Expected behavior
No warning about ancestors

### pylint --version output

Result of `pylint --version` output:
```
pylint 2.7.2
astroid 2.5.1
Python 3.9.1
```

This behaviour has been introduced with pylint 2.7.2. pylint 2.7.1 does not show this warning.

**Comment by Pierre-Sassoulas** [comment]
I think this is due to https://github.com/PyCQA/astroid/pull/916. Ie Mutable set has a long inheritance tree already ` "V => MutableSet => Set => Collection => Sized => Iterable => Container => Generic => object"`. We should probably start counting at the first non builtin type for `too-many-ancestor`.

**Comment by winstxnhdw** [comment]
Hey, I am getting this issue but with `TypedDict` from `typing_extensions`..

**Comment by winstxnhdw** [comment]
Happily #9167
